src/infrastructure/fragmenter.rs

```diff
@@ -159,28 +148,17 @@ impl Fragmentation {
     }
 
     /// If fragment does not exist we need to insert a new entry.
-    fn create_fragment_if_not_exists(&mut self, fragment_header: &FragmentHeader) -> Result<()> {
+    fn create_fragment_if_not_exists(&mut self, fragment_header: FragmentHeader) {
         if !self.fragments.exists(fragment_header.sequence()) {
-            if fragment_header.id() == 0 {
-                match fragment_header.packet_header() {
-                    Some(_header) => {
-                        let reassembly_data = ReassemblyData::new(
-                            fragment_header.sequence(),
-                            fragment_header.fragment_count(),
-                            (9 + self.config.fragment_size) as usize,
-                        );
-
-                        self.fragments
-                            .insert(reassembly_data.clone(), fragment_header.sequence());
-                    }
-                    None => Err(FragmentErrorKind::PacketHeaderNotFound)?,
-                }
-            } else {
-                Err(FragmentErrorKind::AlreadyProcessedFragment)?
-            }
-        }
+            let reassembly_data = ReassemblyData::new(
+                fragment_header.sequence(),
+                fragment_header.fragment_count(),
+                (9 + self.config.fragment_size) as usize,
```

Comment:
Probably a comment on what the 9 is for, or using a constant, would help clarity here

Reply:
Agreed. This should be a constant.